pkg/api/grpcServer.go

```diff
@@ -133,8 +135,11 @@ func (s *grpcServer) dispatcher(stream grpc.Stream, methodName string,
 			worker(ctx2, requestBag, responseBag, request, response)
 
 			sendLock.Lock()
-			tracker.GetResponseAttributes(responseBag, responseAttrs)
-			err := s.sendMsg(stream, response)
+			if err = respTracker.ApplyBag(responseBag, 0, responseAttrs); err != nil {
```

Comment:
That's not how the context stuff is intended to work.

The context is intended to let the sender have different buckets of state held by the other side. A single message results in only one context's attributes being affected. If the mixer can readily identify different kinds of traffic and select between different contexts on the fly, then it could potentially minimize the amount of delta traffic produced. 

The idea was that the proxy could potentially use this to represent the traffic coming from different sources.  

Each context used costs memory on the other side to track a distinct set of attributes. So there must be a constraint and contexts must be reused as a result.

The mixer doesn't have what it needs right now to make an intelligent context choice, and the proxy currently avail itself of this feature either.